test/test_log.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright (C) 2021 Gabriel Ferreira
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+from collections import deque
+
+from test.picardtestcase import PicardTestCase
+
+from picard.log import _calculate_bounds
+
+
+class Item():
+    _log_queue = deque(maxlen=10)
+
+    def __init__(self, pos=0):
+        self.pos = pos
+
+    @staticmethod
+    def contents(prev=-1):
+        if len(Item._log_queue) == 0:
+            return []
+        bounds = _calculate_bounds(prev, Item._log_queue[0].pos, Item._log_queue[-1].pos, len(Item._log_queue))
+
+        contents = [Item._log_queue[i] for i in range(*bounds)]
+        return contents
+
+    @staticmethod
+    def push(item):
+        Item._log_queue.append(item)
+
+    @staticmethod
+    def clear():
+        Item._log_queue.clear()
+
+
+class LogQueueBoundsTestCase(PicardTestCase):
+    def test_1(self):
+        # Common case where the item positions are within the max size of the queue
+        # [0,1,2,3,4,5,6,7], len = 8, maxlen = 10, offset = 0
+        for i in range(8):
+            Item.push(Item(i))
```

Comment:
The tests all use a single global `deque`. I'd rather have a separate instance created in `setup()` for this to initialize a fresh instance for each test. That also removes the need for each test to run `Item.clear()`.